### PR TITLE
Max working branch

### PR DESCRIPTION
**Pull request by devmaxde**
## Checklists

- [ ] I referenced the according Issue
  - [ ] I created an Issue
  - [ ] I put a link in the Section [Related Issue](#related-issue)
- [ ] I tested my code with multiple browsers
- [ ] My commit messages are good and understandable

### CODE REVIEW

**Comment by adrian-goe** [comment]
It might be better to create credetials etc. as environment variables. These can then be exchanged depending on the dockerfile. So hardcoded admin access...

**Review by TheMorpheus407 — CHANGES_REQUESTED** [review]
Good bot!
But you need to include the environment variables in docker, too

File: e_edu_bot/Reactions.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
+import discord
```
Comment (felbinger) [inline e_edu_bot/Reactions.py]: File name should be all lowercase.

File: e_edu_bot/Commands.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
+import discord
```
Comment (felbinger) [inline e_edu_bot/Commands.py]: File name should be all lowercase.

File: docker-compose.yml [hunk 3]
```diff
@@ -1,3 +1,7 @@
-# Please fill out with content
+version: '3'
 
-version: '3.7'
\ No newline at end of file
+services:
+  e_edu_bot:
+    image: #Image Name / Path
+    restart: unless-stopped
+    container_name: e_edu_bot
```
Comment (felbinger) [inline docker-compose.yml:11]: Missing `DISCORD_TOKEN` environment variable:
```yml
environment:
  DISCORD_TOKEN: SECRET_TOKEN
```

File: requirements.txt [hunk 4]
```diff
@@ -0,0 +1 @@
+discord
```
Comment (felbinger) [inline requirements.txt]: Consider adding the required version number.

File: bot.py [hunk 5]
```diff
@@ -0,0 +1,19 @@
+import os
+from e_edu_bot import EEduBot
+import threading
+
+token = os.environ.get("DISCORD_TOKEN")
+
+if token is None or len(token) < 5:
+    raise Exception("Discord Token is not valid")
+
+
```
Comment (felbinger) [inline bot.py]: remove spaces.

File: bot.py [hunk 6]
```diff
@@ -0,0 +1,19 @@
+import os
+from e_edu_bot import EEduBot
+import threading
+
+token = os.environ.get("DISCORD_TOKEN")
+
+if token is None or len(token) < 5:
```
Comment (felbinger) [inline bot.py:7]: Check if file has been execute or imported:
```python
if __name__ == "__main__":
    pass
    # replace pass with your code
```

File: e_edu_bot/config.py [hunk 7]
```diff
@@ -0,0 +1,123 @@
+import json
+import os
+
+
```
Comment (felbinger) [inline e_edu_bot/config.py:4]: Consider creating multiple `Config` classes like `DevelopmentConfig`, `TestingConfig` and `ProductiveConfig` which all extend `Config` to assign yourself to admin_client in the development szenario only.

File: README.md [hunk 8]
```diff
@@ -1,8 +1,50 @@
-# TODO
-Fill out Dockerfile
 
-Fill out docker-compose.yml
+# Python Flask Report Microservice
```
Comment (felbinger) [inline README.md]: ?

File: Dockerfile [hunk 9]
```diff
@@ -1 +1,10 @@
-# Please fill out with content
\ No newline at end of file
+FROM python:3.7-alpine
```
Comment (felbinger) [inline Dockerfile:1]: Use latest image, if possible.

File: config/user_id.json [hunk 10]
```diff
@@ -0,0 +1 @@
+{"104629660800782336": "DevMax#4016", "DevMax#4016": 104629660800782336, "<Pady/think>#2212": 366970301097181195}
```
Comment (felbinger) [inline config/user_id.json]: remove redundant data
`"104629660800782336"`: `"DevMax#4016"`, 
`"DevMax#4016"`: `104629660800782336`

File: e_edu_bot/Commands.py [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+allowed_roles = []
+
+
+async def command_help(client, message: discord.message.Message):
+    # Todo New Help
+    await message.channel.send("Für die Rolle Watcher tippe .+ Watcher #frontend")
+
+
+async def send_welcome_message_id(client, message):
+    text_channel = client.get_channel('1234567890')
```
Comment (felbinger) [inline e_edu_bot/Commands.py]: should be configurable using config

**Review by felbinger — CHANGES_REQUESTED** [review]

File: config/user_id.json (hunk 10 above)
Comment (devmaxde) [reply config/user_id.json]: die erste reihe wurde nur wegen einem bug erstellt.

Comment (devmaxde) [reply config/user_id.json]: Ich habe es ausversehene comitted. eigendlich sollte auf github keine config.json existieren

File: e_edu_bot/config.py (hunk 7 above)
Comment (devmaxde) [reply e_edu_bot/config.py:4]: Changed to example data! Serveradmin need to change this in the generated config File

File: bot.py (hunk 6 above)
Comment (devmaxde) [reply bot.py:7]: ? the file bot.py should not be imported. it only starts the bot

**Pull request merged**